tests/clv/models/test_beta_geo.py

```diff
@@ -172,15 +212,21 @@ def test_expected_probability_alive(self):
         r = 4
 
         recency, frequency, alive, T = self.generate_data(a, b, alpha, r, N, rng=rng)
-        customer_id = list(range(N))
 
+        customer_id = list(range(N))
+        data = pd.DataFrame(
+            {
+                "customer_id": customer_id,
+                "frequency": frequency,
+                "recency": recency,
+                "T": T,
+            }
+        )
         bg_model = BetaGeoModel(
-            customer_id=customer_id,
-            frequency=frequency,
-            recency=recency,
-            T=T,
+            data=data,
         )
-
+        bg_model.build_model()
+        bg_model.fit(fit_method="map")
```

Comment:
Why is this fit call needed?

Reply:
fixed, added extra check in @fit_result.setter and now if self.idata doesn't exist yet it just assigns it, instead of complaining

Reply:
so right now the fit is no longer needed (for the sake of tests of course)